torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2660,6 +2688,8 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         and post-forward are called explicitly, while for the non-recursive-
         wrapping path, they are registered as hooks on every (sub)module.
         """
+        self._lazy_init()
+        self._wait_for_previous_optim_step()
```

Comment:
Let us move this down to be right under the `with torch.autograd.profiler.record_function(...)`?